### ISSUE TITLE
Suffix search

### ISSUE THREAD

**Issue by ouassimBenMosbah** [lucaong/minisearch#51]
Hello, I would like to know if a suffix search exists, or is there any easy "hack" to implement it.

I have a use case where I search through documents having the same starting codes the only differences between them are the few last characters, in order to avoid typing the many first characters that won't help I would like to type only the few last ones.

Here is a stackblitz showing my use case [https://stackblitz.com/edit/angular-ivy-ygmjz9](https://stackblitz.com/edit/angular-ivy-ygmjz9)

**Comment by lucaong** [comment]
Hi @ouassimBenMosbah ,
this feature does not come "out of the box", but it is possible to implement.

One possibility relies on indexing the code field "in reverse", like I did [here](https://angular-ivy-z1wilv.stackblitz.io). That works, but it makes it possible to search only by suffix, or by exact match, while prefix search does not work anymore.

If you want to enable both prefix and suffix search, you might instead do [something like this](https://angular-ivy-m7kzic.stackblitz.io), where I use a custom `extractTerm` function to implement a "virtual field" called `reverseCode`, and I supply a custom `tokenize` upon search, where I produce every term together with its reverse (the specific custom tokenizer I used in the example only works because the `code` field is composed of a single term, otherwise the `tokenize` function would be slightly more complex, but still feasible).

I hope this helps!

**Comment by ouassimBenMosbah** [comment]
Hello @lucaong thanks a lot for your help, it really helped me 😃. Starting from your code I made the prefix and suffix work even for nested fields. [working example](https://stackblitz.com/edit/angular-ivy-zdnqtm)

Now that I have made this working, I am struggling to use the ```combineWith``` option. Indeed, I am feeling that I have "broken" the ```combineWith: 'AND'``` option. Can I ask you for a little help on that one ?
